tests/computedInputs.test.ts

```diff
@@ -0,0 +1,259 @@
+      dmmf,
+      locallyComputedInputs: {
+        createdWithBrowser: ({ ctx }) => ctx.browser,
+      },
+    }),
+  ).toStrictEqual({ data: { name: 'New User', createdWithBrowser: 'firefox' } })
+})
+
+it('removes global computedInputs from all input types', async () => {
+  const { datamodel, ...resolvers } = globalTestData
+  const result = await generateSchemaAndTypes(
+    datamodel,
+    Object.values(resolvers),
+    {
+      globallyComputedInputs: { createdWithBrowser: ({ ctx }) => ctx.browser },
+    },
+  )
+
+  expect(result).toMatchSnapshot('globallyComputedInputs')
+})
+
+it('infers the value of global computedInputs at runtime', async () => {
+  const { datamodel } = globalTestData
+  const dmmf = await getDmmf(datamodel, {
+    globallyComputedInputs: { createdWithBrowser: ({ ctx }) => ctx.browser },
+  })
+  expect(
+    addComputedInputs({
+      params: {
+        root: null,
+        args: {
+          data: {
+            name: 'New User',
+            nested: { create: { name: 'Nested Name' } },
+          },
+        },
+        ctx: { browser: 'firefox' },
+      },
+      inputType: dmmf.getInputType('UserCreateInput'),
+      dmmf,
+      locallyComputedInputs: {},
+    }),
+  ).toStrictEqual({
+    data: {
+      name: 'New User',
+      createdWithBrowser: 'firefox',
+      nested: {
+        create: { createdWithBrowser: 'firefox', name: 'Nested Name' },
+      },
+    },
+  })
+})
+
+it('handles arrays when recursing for computedInputs', async () => {
+  const { datamodel } = globalTestData
+  const dmmf = await getDmmf(datamodel, {
+    globallyComputedInputs: { createdWithBrowser: ({ ctx }) => ctx.browser },
+  })
+
+  expect(
+    addComputedInputs({
+      params: {
+        root: null,
+        args: {
+          data: {
+            name: 'New User',
+            nested: {
+              create: [{ name: 'Nested Name' }, { name: 'Nested Name' }],
+            },
+          },
+        },
+        ctx: { browser: 'firefox' },
+      },
+      inputType: dmmf.getInputType('UserCreateInput'),
+      dmmf,
+      locallyComputedInputs: {},
+    }),
+  ).toStrictEqual({
+    data: {
+      name: 'New User',
+      createdWithBrowser: 'firefox',
+      nested: {
+        create: [
+          { createdWithBrowser: 'firefox', name: 'Nested Name' },
+          { createdWithBrowser: 'firefox', name: 'Nested Name' },
+        ],
+      },
+    },
+  })
+})
+
+it('can combine resolver-level (shallow) and global (deep) computed inputs', async () => {
+  const { datamodel } = globalTestData
+  const dmmf = await getDmmf(datamodel, {
+    // These are applied globally
+    globallyComputedInputs: { createdWithBrowser: ({ ctx }) => ctx.browser },
+  })
+  expect(
+    addComputedInputs({
+      params: {
+        root: null,
+        // name should be required when creating Nested since the computedInput providing
+        // it is specific to UserCreateInput and therefore shallow
```

Comment:
To be honest it was mostly to avoid having to recursively create a new set of types. With the current implementation of the feature, we never create any new types, we just modify existing ones, but that would cause collisions if we allowed deep local computedInputs. Totally open to implementing that now that I have a better understanding of the codebase and especially since that's already how https://github.com/prisma-labs/nexus-prisma/pull/566 works, but would definitely prefer to make that an option in a later PR if possible.

Reply:
Yep its a clear limitation that we can address as an isolated feature later on.